monkey/agent_plugins/exploiters/smb/src/smb_exploit_client.py

```diff
@@ -1,264 +1,177 @@
 import logging
-import ntpath
-from io import BytesIO
 from pathlib import PurePath
-from typing import Any, Dict, Optional, Sequence, Tuple
-
-from impacket.dcerpc.v5 import scmr, srvs, transport
-from impacket.smbconnection import SMBConnection
+from typing import List, Set, Tuple, Type
 
 from common import OperatingSystem
 from common.credentials import Credentials
-from common.event_queue import IAgentEventPublisher
-from common.types import Event
-from infection_monkey.exploit.tools import RemoteFileCopyError
-from infection_monkey.exploit.tools.helpers import get_agent_dst_path
+from common.tags import (
+    T1021_ATTACK_TECHNIQUE_TAG,
+    T1105_ATTACK_TECHNIQUE_TAG,
+    T1110_ATTACK_TECHNIQUE_TAG,
+    T1135_ATTACK_TECHNIQUE_TAG,
+    T1210_ATTACK_TECHNIQUE_TAG,
+    T1569_ATTACK_TECHNIQUE_TAG,
+)
+from common.types import NetworkPort
+from infection_monkey.exploit.tools import (
+    IRemoteAccessClient,
+    RemoteAuthenticationError,
+    RemoteCommandExecutionError,
+    RemoteFileCopyError,
+)
 from infection_monkey.i_puppet import TargetHost
 
-from .smb_command_builder import build_smb_command
+from .smb_client import ShareInfo, SMBClient
 from .smb_options import SMBOptions
```

Comment:
```suggestion
from smb_client import ShareInfo, SMBClient
from smb_options import SMBOptions
```

Reply:
^ can we convert to absolute imports if we make it a package?

Reply:
No need to do that

Reply:
absolute imports > local

Reply:
> They also have a benefit that you don't need to modify them when moving files
That depends on where you move them to.

Reply:
>No, absolute imports imply package scope (if you take a look at site-package none of them use local imports). 

@VakarisZ I'm not sure what you mean:
![image](https://user-images.githubusercontent.com/19957806/227261574-62a910be-06dc-4755-a629-5a6a8e2af7a7.png)